picard/util/tags.py

```diff
@@ -40,186 +40,732 @@
+        is_preserved=True,
+        is_tag=False,
+    ),
+    TagVar(
+        'website',
+        shortdesc=N_('Artist Website'),
+    ),
+    TagVar(
+        'work',
+        shortdesc=N_('Work'),
+    ),
+    TagVar(
+        'writer',
+        shortdesc=N_('Writer'),
+    ),
 }
 
-PRESERVED_TAGS = (
-    '~bitrate',
-    '~bits_per_sample',
-    '~channels',
-    '~dirname',
-    '~extension',
-    '~filename',
-    '~file_created_timestamp',
-    '~file_modified_timestamp',
-    '~format',
-    '~sample_rate',
-    '~video',
-)
+ALL_NAME_2_TAGVAR = {str(tv): tv for tv in ALL_TAG_VARS}
+
+
+def tag_names():
+    yield from (str(tv) for tv in ALL_TAG_VARS if tv.is_tag)
+
+
+PRESERVED_TAGS = tuple(str(tv) for tv in ALL_TAG_VARS if tv.is_preserved)
 
 # Tags that got generated in some way from the audio content.
 # Those can be set by Picard but the new values usually should be kept
 # when moving the file between tags.
-CALCULATED_TAGS = {
-    'acoustid_fingerprint',
-    'acoustid_id',
-    'replaygain_album_gain',
-    'replaygain_album_peak',
-    'replaygain_album_range',
-    'replaygain_reference_loudness',
-    'replaygain_track_gain',
-    'replaygain_track_peak',
-    'replaygain_track_range',
-    'r128_album_gain',
-    'r128_track_gain',
-}
+CALCULATED_TAGS = set(str(tv) for tv in ALL_TAG_VARS if tv.is_calculated)
 
 # Tags that contains infos related to files
-FILE_INFO_TAGS = {
-    '~bitrate',
-    '~bits_per_sample',
-    '~channels',
-    '~filesize',
-    '~format',
-    '~sample_rate',
-}
+FILE_INFO_TAGS = set(str(tv) for tv in ALL_TAG_VARS if tv.is_file_info)
 
 # Variables available to scripts (used by script editor completer)
-EXTRA_VARIABLES = (
-    '~absolutetracknumber',
-    '~albumartists_countries',
-    '~albumartists_sort',
-    '~albumartists',
-    '~artists_countries',
-    '~artists_sort',
-    '~datatrack',
-    '~discpregap',
-    '~multiartist',
-    '~musicbrainz_discids',
-    '~musicbrainz_tracknumber',
-    '~performance_attributes',
-    '~pregap',
-    '~primaryreleasetype',
-    '~rating',
-    '~recording_firstreleasedate',
-    '~recordingcomment',
-    '~recordingtitle',
-    '~releasecomment',
-    '~releasecountries',
-    '~releasegroup_firstreleasedate',
-    '~releasegroup',
-    '~releasegroupcomment',
-    '~releaselanguage',
-    '~secondaryreleasetype',
-    '~silence',
-    '~totalalbumtracks',
-    '~video',
-)
+SCRIPT_VARIABLES = tuple(str(tv) for tv in ALL_TAG_VARS if tv.is_script_variable)
 
 
 def display_tag_name(name):
+    tagdesc = None
     if ':' in name:
-        name, desc = name.split(':', 1)
-        if desc:
-            return '%s [%s]' % (_(TAG_NAMES.get(name, name)), desc)
-    return _(TAG_NAMES.get(name, name))
+        name, tagdesc = name.split(':', 1)
+
+    tagvar = ALL_NAME_2_TAGVAR.get(name, None)
+    tagtitle = _(tagvar.shortdesc) if tagvar else name
+
+    if tagdesc:
+        return '%s [%s]' % (tagtitle, tagdesc)
+    else:
+        return tagtitle
```

Comment:
This is good for the column headers, but I assume you would have a similar function returning the long descriptions for the tooltips?